checkbox-provider-ce-oem/bin/tcp_multi_connections.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python3
+import socket
+import argparse
+import threading
+import logging
+import time
+import string
+import random
+from datetime import datetime, timedelta
+from enum import Enum
+
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[
+        logging.StreamHandler(),
+    ],
+)
+
+
+class StatusEnum(Enum):
+    SUCCESS = 0
+    ERROR = 1
+    FAIL = 2
+
+
+def generate_random_string(length):
+    letters_and_digits = string.ascii_letters + string.digits
+    random_string = ''.join(
+        random.choice(letters_and_digits) for _ in range(length))
+    return random_string
+
+
+def sorting_data(dict_status):
+    list_times = []
+    fail_records = []
+    for value in dict_status.values():
+        list_times.append(value.get('time'))
+        if not value.get('status'):
+            fail_records.append(value)
+    return list_times, fail_records
+
+
+def format_output(port, message="", dict_status={}):
+    status = StatusEnum.SUCCESS
+    times = records = None
+    if message:
+        status = StatusEnum.ERROR
+    elif dict_status:
+        times, records = sorting_data(dict_status)
+        if records:
+            status = StatusEnum.FAIL
+            message = "Received payload incorrect!"
+        else:
+            message = "Received payload correct!"
+
+    return {
+        'port': port,
+        'status': status,
+        'message': message,
+        'fail': records if records else None,
+        'total_period': sum(times, timedelta()) if times else None,
+        'avg_period': sum(times, timedelta())/len(times) if times else None,
+        'max_period': max(times) if times else None,
+        'min_period': min(times) if times else None,
+    }
+
+
+def generate_result(results, run_time):
```

Comment:
the function name is a bit wired to me, because of we did not generate any data here.
Probably we could named as check_results